langchain/document_loaders/directory.py

```diff
@@ -36,31 +29,38 @@ def __init__(
         loader_kwargs: Union[dict, None] = None,
         recursive: bool = False,
     ):
-        """Initialize with path to directory and how to glob over it."""
-        if loader_kwargs is None:
-            loader_kwargs = {}
-        self.path = path
-        self.glob = glob
-        self.load_hidden = load_hidden
+        """Initialize with path to directory and how to glob over it.
+
+        Args:
+            path: Path to directory.
+            glob: Glob pattern to use.
+            silent_errors: If True, errors will be logged and ignored.
+            load_hidden: If True, hidden files will be loaded.
+            loader_cls: Class to use for loading files.
+            loader_kwargs: Keyword arguments to pass to loader_cls.
+            recursive: If True, will recursively load files.
+        """
+        self.loader = FileSystemLoader(
+            path, glob, load_hidden=load_hidden, recursive=recursive
+        )
         self.loader_cls = loader_cls
-        self.loader_kwargs = loader_kwargs
+        self.loader_kwargs = loader_kwargs or {}
         self.silent_errors = silent_errors
-        self.recursive = recursive
 
     def load(self) -> List[Document]:
         """Load documents."""
-        p = Path(self.path)
-        docs = []
-        items = p.rglob(self.glob) if self.recursive else p.glob(self.glob)
-        for i in items:
-            if i.is_file():
-                if _is_visible(i.relative_to(p)) or self.load_hidden:
-                    try:
-                        sub_docs = self.loader_cls(str(i), **self.loader_kwargs).load()
-                        docs.extend(sub_docs)
-                    except Exception as e:
-                        if self.silent_errors:
-                            logger.warning(e)
-                        else:
-                            raise e
-        return docs
+        return list(self.lazy_load())
+
+    def lazy_load(
+        self,
+    ) -> Generator[Document, None, None]:
+        """Load documents lazily."""
+        for blob in self.loader.yield_blobs():
+            try:
+                sub_docs = self.loader_cls(blob.data, **self.loader_kwargs).load()
+                yield from sub_docs
+            except Exception as e:
+                if self.silent_errors:
+                    logger.warning(e)
```

Comment:
ooc should errors (even if silenced) use logger.error?

Reply:
yeah we should update to logger.error probably or allow controlling the level